### PR TITLE
v1.18: Scheduler - prioritization fees/cost (#34888)

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Manual backport of https://github.com/solana-labs/solana/pull/34888

#### Summary of Changes
- Removed changes for prioritization metrics since those do not exist in 1.18

Fixes #

### CODE REVIEW

**Review by tao-stones — DISMISSED** [review]
lgtm

File: core/src/banking_stage/transaction_scheduler/transaction_state.rs [hunk 1]
```diff
@@ -34,77 +30,38 @@ pub(crate) enum TransactionState {
...
         Self::Unprocessed {
             transaction_ttl,
-            transaction_priority_details,
-            transaction_cost,
-            forwarded: false,
-        }
-    }
-
-    /// Returns a reference to the priority details of the transaction.
-    pub(crate) fn transaction_priority_details(&self) -> &TransactionPriorityDetails {
```
Comment (t-nelson) [inline core/src/banking_stage/transaction_scheduler/transaction_state.rs:65]: for the sake of backport readability, can you read these removed functions and annotate with `#[allow(dead_code)]`?

Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/transaction_state.rs:65]: The meta data stored in transaction_state changed, so we no longer have a `TransactionPriorityDetails`. I don't think we should add back those unused metadata and code for calculating them.

I could add a block-comment around these removed fns if you think that'd help simplify the diff.

Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/transaction_state.rs:65]: Added block-comment from 50 to 97: c80d5fb2e9fcae399e451894dfcda470f8a1c0dd

**Comment by apfitzge** [comment]
had to rebase due to 1.18 version being tagged, was causing sanity CI check to fail.

**Review by t-nelson — APPROVED** [review]
great! much more readable with the dead code commented instead of removed

**Pull request merged**